src/topologies/halfedge.jl

```diff
@@ -208,52 +206,60 @@ function HalfEdgeTopology(elems::AbstractVector{<:Connectivity}; sort=true)
   halves = Vector{Tuple{HalfEdge,HalfEdge}}()
   visited = Set{Tuple{Int,Int}}()
   for ((u, v), he) in half4pair
-    if (u, v) ∉ visited
+    if minmax(u, v) ∉ visited
       push!(halves, (he, he.half))
-      push!(visited, (u, v))
-      push!(visited, (v, u))
+      push!(visited, minmax(u, v))
     end
   end
 
   HalfEdgeTopology(halves)
 end
 
-function adjsort(elems::AbstractVector{<:Connectivity})
+function adjsortperm(elems::AbstractVector{<:Connectivity})
   # initialize list of adjacent elements
   # with first element from original list
-  list = indices.(elems)
-  adjs = Tuple[popfirst!(list)]
-
-  # the loop will terminate if the mesh
-  # is manifold, and that is always true
-  # with half-edge topology
-  while !isempty(list)
-    # lookup all elements that share at least
-    # one vertex with the last adjacent element
-    found = false
-    vinds = last(adjs)
-    for i in vinds
-      einds = findall(e -> i ∈ e, list)
-      if !isempty(einds)
-        # lookup all elements that share at
-        # least two vertices (i.e. edge)
-        for j in sort(einds, rev=true)
-          if length(vinds ∩ list[j]) > 1
-            found = true
-            push!(adjs, popat!(list, j))
-          end
+  einds = Int[1]
+  sizehint!(einds, length(elems))
+
+  # remaining list of elements to process
+  oinds = collect(2:length(elems))
+
+  # lookup all elements that share at least
+  # two vertices (i.e., edge) with the last
+  # adjacent element
+  found = false
+  while !isempty(oinds)
+    lelem = elems[last(einds)]
+    vinds = indices(lelem)
+    for v in vinds
+      # vertices that are not `v`
+      v! = filter(!=(v), vinds)
+
+      # iteratively test other elements
+      iter = 1
+      while iter ≤ length(oinds)
+        oelem = elems[oinds[iter]]
+        vinds′ = indices(oelem)
+        if any(==(v), vinds′) && !isdisjoint(v!, vinds′)
+          found = true
+          push!(einds, popat!(oinds, iter))
+        else
+          found = false
+          iter += 1
+          continue
```

Comment:
This `continue` has effect? Given that it is the last statement in the `while` loop, I think we can erase it?